packages/flutter/lib/src/material/slider_theme.dart

```diff
@@ -927,10 +902,132 @@ class _EmptySliderComponentShape extends SliderComponentShape {
+  /// [SliderThemeData.trackHeight].
+  ///
+  /// The [Rect] is centered both horizontally and vertically within the slider
+  /// bounds.
+  Rect getPreferredRect({
+    RenderBox parentBox,
+    Offset offset = Offset.zero,
+    SliderThemeData sliderTheme,
+    bool isEnabled = false,
+    bool isDiscrete = false,
+  }) {
+    final double thumbWidth = sliderTheme.thumbShape.getPreferredSize(isEnabled, isDiscrete).width;
+    final double overlayWidth = sliderTheme.overlayShape.getPreferredSize(isEnabled, isDiscrete).width;
+    final double trackHeight = sliderTheme.trackHeight;
+    assert(overlayWidth >= 0);
+    assert(trackHeight >= 0);
+    assert(parentBox.size.width >= overlayWidth);
+    assert(parentBox.size.height >= trackHeight);
+
+    final double trackLeft = offset.dx + overlayWidth / 2;
+    final double trackTop = offset.dy + (parentBox.size.height - trackHeight) / 2;
+    final double trackWidth = parentBox.size.width - math.max(thumbWidth, overlayWidth);
+    return Rect.fromLTWH(trackLeft, trackTop, trackWidth, trackHeight);
+  }
+}
 
 /// This is the default shape of a [Slider]'s track.
 ///
+/// It paints a solid colored rectangle with rounded edges, vertically centered
+/// in the [parentBox]. The track rectangle extends to the bounds of the
+/// [parentBox], but is padded by the larger of [RoundSliderOverlayShape]
+/// radius and [RoundSliderThumbShape]. The height is defined by the
+/// [SliderThemeData.trackHeight]. The color is determined by the [Slider]'s
+/// enabled state and the track piece's active state which are defined by:
+///   [SliderThemeData.activeTrackColor],
+///   [SliderThemeData.inactiveTrackColor],
+///   [SliderThemeData.disabledActiveTrackColor],
+///   [SliderThemeData.disabledInactiveTrackColor].
+///
+/// See also:
+///
+///  * [Slider] for the component that this is meant to display this shape.
+///  * [SliderThemeData] where an instance of this class is set to inform the
+///    slider of the visual details of the its track.
+///  * [SliderTrackShape] Base component for creating other custom track
+///    shapes.
+///  * [RectangularSliderTrackShape] for a similar track with blunt edges.
+class RoundedRectSliderTrackShape extends SliderTrackShape with BaseSliderTrackShape {
+  /// Create a slider track that draws 2 rectangles with rounded outer edges.
+  const RoundedRectSliderTrackShape();
+
+  @override
+  void paint(
+    PaintingContext context,
+    Offset offset, {
+    RenderBox parentBox,
+    SliderThemeData sliderTheme,
+    Animation<double> enableAnimation,
+    TextDirection textDirection,
+    Offset thumbCenter,
+    bool isDiscrete,
+    bool isEnabled,
+  }) {
+    // If the slider track height is 0, then it makes no difference whether the
+    // track is painted or not, therefore the painting can be a no-op.
+    if (sliderTheme.trackHeight == 0) {
+      return;
+    }
+
+    // Assign the track segment paints, which are left: active, right: inactive,
```

Comment:
Are the left and right in RTL?

Reply:
And I see one line below that they are. 

Change this to leading and trailing instead of right and left and you get all that for free.

Reply:
I mean the comments will be clear without having to say what happens in RTL.

Reply:
It is noted that it is reversed for RTL on the next line